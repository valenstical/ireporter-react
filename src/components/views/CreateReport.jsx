import React from 'react';
import PropTypes from 'prop-types';
import loadingIcon from '../../../public/images/resources/loader-light.gif';
import Alert from './Alert';

/**
 * The create report component
 * @returns {object} The generated JSX object
 */
export default function CreateReport({ handleSubmit, report }) {
  return (
    <section className="login-wrapper top-space account-wrapper" data-pg-name="Dashboard">
      <section className="heading">
        <div className="wrapper" data-pg-name="Wrapper">
          <div className="max-width-800">
            <div className="bread">
              <h1>Report an Incident</h1>
            </div>
          </div>
        </div>
      </section>
      <div className="wrapper" data-pg-name="Wrapper">
        <form onSubmit={handleSubmit} noValidate className="max-width-800">
          {report.message.length > 0 && <Alert message={report.message} title={report.success ? 'Report Successful!' : 'Oops!'} success={report.success} />}
          <div className="report-item card card-sm create-report-details">
            <div className="form-wrapper">
              <label htmlFor="title">Title </label>
              <small className="text-helper">(e.g Power outage for over 3 weeks around Aba market)</small>
              <input id="title" className="form-element" type="text" name="title" required defaultValue={report.title} />
            </div>
            <div className="form-wrapper no-margin-btm">
              <label htmlFor="comments">
Comment
              </label>
              <small className="text-helper">(Write full details here)</small>
              <textarea id="comments" className="form-element" name="comment" required defaultValue={report.comment} />
            </div>
          </div>
          <div className="report-item card card-sm create-report-details padding-zero-btm">
            <h2 className="report-box-title">Report Type</h2>
            <p>
Is this incident linked to corruption or
              just an issue that needs government intervention ?

            </p>
            <div className="row" data-pg-name="Row">
              <div className="column column-md-6" data-pg-name="Column">
                <div className="report-item report-box card no-margin-btm" title="Red Flag Report">
                  <h4>
Corruption case
                    {' '}
                    <i className="fa fa-flag flag" title="Red Flag Report" />
                  </h4>
                  <p>e.g. bribery, extortion, money laundering, e.t.c</p>
                  <input type="radio" name="type" defaultValue="red-flag" data-pg-hidden className="radio-fine" defaultChecked />
                  <div className="pg-empty-placeholder radio-mark" />
                </div>
              </div>
              <div className="column column-md-6" data-pg-name="Column">
                <div className="report-item report-box card" title="Intervention Report">
                  <h4>
Intervention Case
                    {' '}
                    <i className="fa flag fa-bullhorn flag-intervention" title="Intervention Report" />
                  </h4>
                  <p>e.g. bad road, collapsed bridge, flooding, e.t.c</p>
                  <input type="radio" name="type" defaultValue="intervention" className="radio-fine" data-pg-hidden defaultChecked={report.type === 'intervention'} />
                  <div className="pg-empty-placeholder radio-mark" />
                </div>
              </div>
            </div>
          </div>
          <div className="report-item card card-sm max-width-800 create-report-details padding-zero-btm">
            <header className="header-create">
              <h2 className="report-box-title">Location</h2>
              <p>
Where did this happen? Enter the the complete location of the incident
              </p>
            </header>
            <div className="row" data-pg-name="Row">
              <div className="column" data-pg-name="column-md-6">
                <div className="form-wrapper">
                  <input id="location" defaultValue={report.state} className="form-element" type="text" name="state" required />
                </div>
              </div>
            </div>
          </div>
          <div className="btn-wrapper">
            <button disabled={report.isBusy} className="btn-brand btn-250 btn-submit" type="submit">
              <span>Submit Report</span>
              { report.isBusy && <img src={loadingIcon} width="12" alt="loading" />}
            </button>
          </div>
        </form>
      </div>
    </section>

  );
}

CreateReport.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  report: PropTypes.object.isRequired
};
