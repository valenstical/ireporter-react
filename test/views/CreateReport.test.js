import React from 'react';
import { shallow } from 'enzyme';
import CreateReport from '../../src/components/views/CreateReport';

describe('<CreateReport>', () => {
  it('should render without crashing for red-flag report', () => {
    const wrapper = shallow(<CreateReport
handleSubmit={jest.fn()} report={{
  message: [],
  success: false,
  isBusy: false,
  type: 'red-flag'
}} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render without crashing for intervention report', () => {
    const wrapper = shallow(<CreateReport
handleSubmit={jest.fn()} report={{
  message: [],
  success: false,
  isBusy: false,
  type: 'intervention'
}} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render without crashing for successful report', () => {
    const wrapper = shallow(<CreateReport
handleSubmit={jest.fn()} report={{
  message: ['successful'],
  success: true,
  isBusy: false,
  type: 'intervention'
}} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render without crashing for failed report', () => {
    const wrapper = shallow(<CreateReport
handleSubmit={jest.fn()} report={{
  message: ['failed'],
  success: false,
  isBusy: true,
  type: 'red-flag'
}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
