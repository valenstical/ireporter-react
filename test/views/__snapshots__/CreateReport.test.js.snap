// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateReport> should render without crashing for failed report 1`] = `
<section
  className="login-wrapper top-space account-wrapper"
  data-pg-name="Dashboard"
>
  <section
    className="heading"
  >
    <div
      className="wrapper"
      data-pg-name="Wrapper"
    >
      <div
        className="max-width-800"
      >
        <div
          className="bread"
        >
          <h1>
            Report an Incident
          </h1>
        </div>
      </div>
    </div>
  </section>
  <div
    className="wrapper"
    data-pg-name="Wrapper"
  >
    <form
      noValidate={true}
      onSubmit={[MockFunction]}
    >
      <Alert
        message={
          Array [
            "failed",
          ]
        }
        success={false}
        title="Oops!"
      />
      <div
        className="report-item card card-sm max-width-800 create-report-details"
      >
        <div
          className="form-wrapper"
        >
          <label
            htmlFor="title"
          >
            Title 
          </label>
          <small
            className="text-helper"
          >
            (e.g Power outage for over 3 weeks around Aba market)
          </small>
          <input
            className="form-element"
            id="title"
            name="title"
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-wrapper no-margin-btm"
        >
          <label
            htmlFor="comments"
          >
            Comment
          </label>
          <small
            className="text-helper"
          >
            (Write full details here)
          </small>
          <textarea
            className="form-element"
            id="comments"
            name="comment"
            required={true}
          />
        </div>
      </div>
      <div
        className="report-item card card-sm max-width-800 create-report-details padding-zero-btm"
      >
        <h2
          className="report-box-title"
        >
          Report Type
        </h2>
        <p>
          Is this incident linked to corruption or just an issue that needs government intervention ?
        </p>
        <div
          className="row"
          data-pg-name="Row"
        >
          <div
            className="column column-md-6"
            data-pg-name="Column"
          >
            <div
              className="report-item max-width-800 report-box card no-margin-btm"
              title="Red Flag Report"
            >
              <h4>
                Corruption case
                 
                <i
                  className="fa fa-flag flag"
                  title="Red Flag Report"
                />
              </h4>
              <p>
                e.g. bribery, extortion, money laundering, e.t.c
              </p>
              <input
                className="radio-fine"
                data-pg-hidden={true}
                defaultChecked={true}
                defaultValue="red-flag"
                name="type"
                type="radio"
              />
              <div
                className="pg-empty-placeholder radio-mark"
              />
            </div>
          </div>
          <div
            className="column column-md-6"
            data-pg-name="Column"
          >
            <div
              className="report-item max-width-800 report-box card"
              title="Intervention Report"
            >
              <h4>
                Intervention Case
                 
                <i
                  className="fa flag fa-bullhorn flag-intervention"
                  title="Intervention Report"
                />
              </h4>
              <p>
                e.g. bad road, collapsed bridge, flooding, e.t.c
              </p>
              <input
                className="radio-fine"
                data-pg-hidden={true}
                defaultChecked={false}
                defaultValue="intervention"
                name="type"
                type="radio"
              />
              <div
                className="pg-empty-placeholder radio-mark"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="report-item card card-sm max-width-800 create-report-details padding-zero-btm"
      >
        <header
          className="header-create"
        >
          <h2
            className="report-box-title"
          >
            Location
          </h2>
          <p>
            Where did this happen? Enter the latitude/longitude. You can use
             
            <Link
              replace={false}
              target="_blank"
              to="https://maps.google.com"
            >
              <strong>
                <ins>
                  Google Map
                </ins>
              </strong>
            </Link>
             
            for help.
          </p>
          <p
            className="hidden"
          >
            Where did this happen? Enter the latitude/longitude or simply use the marker to choose a location from the map
          </p>
        </header>
        <div
          className="row"
          data-pg-name="Row"
        >
          <div
            className="column-md-6 column column-xs-6"
            data-pg-name="column-md-6"
          >
            <div
              className="form-wrapper"
            >
              <label
                htmlFor="latitude"
              >
                Latitude
              </label>
              <input
                className="form-element"
                id="latitude"
                name="latitude"
                required={true}
                type="tel"
              />
            </div>
          </div>
          <div
            className="column-md-6 column column-xs-6"
            data-pg-name="column-md-6"
          >
            <div
              className="form-wrapper"
            >
              <label
                htmlFor="longitude"
              >
                Longitude
              </label>
              <input
                className="form-element"
                id="longitude"
                name="longitude"
                required={true}
                type="tel"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="btn-wrapper max-width-800"
      >
        <button
          className="btn-brand btn-250 btn-submit"
          disabled={true}
          type="submit"
        >
          <span>
            Submit Report
          </span>
          <img
            alt="loading"
            src={Object {}}
            width="12"
          />
        </button>
      </div>
    </form>
  </div>
</section>
`;

exports[`<CreateReport> should render without crashing for intervention report 1`] = `
<section
  className="login-wrapper top-space account-wrapper"
  data-pg-name="Dashboard"
>
  <section
    className="heading"
  >
    <div
      className="wrapper"
      data-pg-name="Wrapper"
    >
      <div
        className="max-width-800"
      >
        <div
          className="bread"
        >
          <h1>
            Report an Incident
          </h1>
        </div>
      </div>
    </div>
  </section>
  <div
    className="wrapper"
    data-pg-name="Wrapper"
  >
    <form
      noValidate={true}
      onSubmit={[MockFunction]}
    >
      <div
        className="report-item card card-sm max-width-800 create-report-details"
      >
        <div
          className="form-wrapper"
        >
          <label
            htmlFor="title"
          >
            Title 
          </label>
          <small
            className="text-helper"
          >
            (e.g Power outage for over 3 weeks around Aba market)
          </small>
          <input
            className="form-element"
            id="title"
            name="title"
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-wrapper no-margin-btm"
        >
          <label
            htmlFor="comments"
          >
            Comment
          </label>
          <small
            className="text-helper"
          >
            (Write full details here)
          </small>
          <textarea
            className="form-element"
            id="comments"
            name="comment"
            required={true}
          />
        </div>
      </div>
      <div
        className="report-item card card-sm max-width-800 create-report-details padding-zero-btm"
      >
        <h2
          className="report-box-title"
        >
          Report Type
        </h2>
        <p>
          Is this incident linked to corruption or just an issue that needs government intervention ?
        </p>
        <div
          className="row"
          data-pg-name="Row"
        >
          <div
            className="column column-md-6"
            data-pg-name="Column"
          >
            <div
              className="report-item max-width-800 report-box card no-margin-btm"
              title="Red Flag Report"
            >
              <h4>
                Corruption case
                 
                <i
                  className="fa fa-flag flag"
                  title="Red Flag Report"
                />
              </h4>
              <p>
                e.g. bribery, extortion, money laundering, e.t.c
              </p>
              <input
                className="radio-fine"
                data-pg-hidden={true}
                defaultChecked={true}
                defaultValue="red-flag"
                name="type"
                type="radio"
              />
              <div
                className="pg-empty-placeholder radio-mark"
              />
            </div>
          </div>
          <div
            className="column column-md-6"
            data-pg-name="Column"
          >
            <div
              className="report-item max-width-800 report-box card"
              title="Intervention Report"
            >
              <h4>
                Intervention Case
                 
                <i
                  className="fa flag fa-bullhorn flag-intervention"
                  title="Intervention Report"
                />
              </h4>
              <p>
                e.g. bad road, collapsed bridge, flooding, e.t.c
              </p>
              <input
                className="radio-fine"
                data-pg-hidden={true}
                defaultChecked={true}
                defaultValue="intervention"
                name="type"
                type="radio"
              />
              <div
                className="pg-empty-placeholder radio-mark"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="report-item card card-sm max-width-800 create-report-details padding-zero-btm"
      >
        <header
          className="header-create"
        >
          <h2
            className="report-box-title"
          >
            Location
          </h2>
          <p>
            Where did this happen? Enter the latitude/longitude. You can use
             
            <Link
              replace={false}
              target="_blank"
              to="https://maps.google.com"
            >
              <strong>
                <ins>
                  Google Map
                </ins>
              </strong>
            </Link>
             
            for help.
          </p>
          <p
            className="hidden"
          >
            Where did this happen? Enter the latitude/longitude or simply use the marker to choose a location from the map
          </p>
        </header>
        <div
          className="row"
          data-pg-name="Row"
        >
          <div
            className="column-md-6 column column-xs-6"
            data-pg-name="column-md-6"
          >
            <div
              className="form-wrapper"
            >
              <label
                htmlFor="latitude"
              >
                Latitude
              </label>
              <input
                className="form-element"
                id="latitude"
                name="latitude"
                required={true}
                type="tel"
              />
            </div>
          </div>
          <div
            className="column-md-6 column column-xs-6"
            data-pg-name="column-md-6"
          >
            <div
              className="form-wrapper"
            >
              <label
                htmlFor="longitude"
              >
                Longitude
              </label>
              <input
                className="form-element"
                id="longitude"
                name="longitude"
                required={true}
                type="tel"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="btn-wrapper max-width-800"
      >
        <button
          className="btn-brand btn-250 btn-submit"
          disabled={false}
          type="submit"
        >
          <span>
            Submit Report
          </span>
        </button>
      </div>
    </form>
  </div>
</section>
`;

exports[`<CreateReport> should render without crashing for red-flag report 1`] = `
<section
  className="login-wrapper top-space account-wrapper"
  data-pg-name="Dashboard"
>
  <section
    className="heading"
  >
    <div
      className="wrapper"
      data-pg-name="Wrapper"
    >
      <div
        className="max-width-800"
      >
        <div
          className="bread"
        >
          <h1>
            Report an Incident
          </h1>
        </div>
      </div>
    </div>
  </section>
  <div
    className="wrapper"
    data-pg-name="Wrapper"
  >
    <form
      noValidate={true}
      onSubmit={[MockFunction]}
    >
      <div
        className="report-item card card-sm max-width-800 create-report-details"
      >
        <div
          className="form-wrapper"
        >
          <label
            htmlFor="title"
          >
            Title 
          </label>
          <small
            className="text-helper"
          >
            (e.g Power outage for over 3 weeks around Aba market)
          </small>
          <input
            className="form-element"
            id="title"
            name="title"
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-wrapper no-margin-btm"
        >
          <label
            htmlFor="comments"
          >
            Comment
          </label>
          <small
            className="text-helper"
          >
            (Write full details here)
          </small>
          <textarea
            className="form-element"
            id="comments"
            name="comment"
            required={true}
          />
        </div>
      </div>
      <div
        className="report-item card card-sm max-width-800 create-report-details padding-zero-btm"
      >
        <h2
          className="report-box-title"
        >
          Report Type
        </h2>
        <p>
          Is this incident linked to corruption or just an issue that needs government intervention ?
        </p>
        <div
          className="row"
          data-pg-name="Row"
        >
          <div
            className="column column-md-6"
            data-pg-name="Column"
          >
            <div
              className="report-item max-width-800 report-box card no-margin-btm"
              title="Red Flag Report"
            >
              <h4>
                Corruption case
                 
                <i
                  className="fa fa-flag flag"
                  title="Red Flag Report"
                />
              </h4>
              <p>
                e.g. bribery, extortion, money laundering, e.t.c
              </p>
              <input
                className="radio-fine"
                data-pg-hidden={true}
                defaultChecked={true}
                defaultValue="red-flag"
                name="type"
                type="radio"
              />
              <div
                className="pg-empty-placeholder radio-mark"
              />
            </div>
          </div>
          <div
            className="column column-md-6"
            data-pg-name="Column"
          >
            <div
              className="report-item max-width-800 report-box card"
              title="Intervention Report"
            >
              <h4>
                Intervention Case
                 
                <i
                  className="fa flag fa-bullhorn flag-intervention"
                  title="Intervention Report"
                />
              </h4>
              <p>
                e.g. bad road, collapsed bridge, flooding, e.t.c
              </p>
              <input
                className="radio-fine"
                data-pg-hidden={true}
                defaultChecked={false}
                defaultValue="intervention"
                name="type"
                type="radio"
              />
              <div
                className="pg-empty-placeholder radio-mark"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="report-item card card-sm max-width-800 create-report-details padding-zero-btm"
      >
        <header
          className="header-create"
        >
          <h2
            className="report-box-title"
          >
            Location
          </h2>
          <p>
            Where did this happen? Enter the latitude/longitude. You can use
             
            <Link
              replace={false}
              target="_blank"
              to="https://maps.google.com"
            >
              <strong>
                <ins>
                  Google Map
                </ins>
              </strong>
            </Link>
             
            for help.
          </p>
          <p
            className="hidden"
          >
            Where did this happen? Enter the latitude/longitude or simply use the marker to choose a location from the map
          </p>
        </header>
        <div
          className="row"
          data-pg-name="Row"
        >
          <div
            className="column-md-6 column column-xs-6"
            data-pg-name="column-md-6"
          >
            <div
              className="form-wrapper"
            >
              <label
                htmlFor="latitude"
              >
                Latitude
              </label>
              <input
                className="form-element"
                id="latitude"
                name="latitude"
                required={true}
                type="tel"
              />
            </div>
          </div>
          <div
            className="column-md-6 column column-xs-6"
            data-pg-name="column-md-6"
          >
            <div
              className="form-wrapper"
            >
              <label
                htmlFor="longitude"
              >
                Longitude
              </label>
              <input
                className="form-element"
                id="longitude"
                name="longitude"
                required={true}
                type="tel"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="btn-wrapper max-width-800"
      >
        <button
          className="btn-brand btn-250 btn-submit"
          disabled={false}
          type="submit"
        >
          <span>
            Submit Report
          </span>
        </button>
      </div>
    </form>
  </div>
</section>
`;

exports[`<CreateReport> should render without crashing for successful report 1`] = `
<section
  className="login-wrapper top-space account-wrapper"
  data-pg-name="Dashboard"
>
  <section
    className="heading"
  >
    <div
      className="wrapper"
      data-pg-name="Wrapper"
    >
      <div
        className="max-width-800"
      >
        <div
          className="bread"
        >
          <h1>
            Report an Incident
          </h1>
        </div>
      </div>
    </div>
  </section>
  <div
    className="wrapper"
    data-pg-name="Wrapper"
  >
    <form
      noValidate={true}
      onSubmit={[MockFunction]}
    >
      <Alert
        message={
          Array [
            "successful",
          ]
        }
        success={true}
        title="Report Successful!"
      />
      <div
        className="report-item card card-sm max-width-800 create-report-details"
      >
        <div
          className="form-wrapper"
        >
          <label
            htmlFor="title"
          >
            Title 
          </label>
          <small
            className="text-helper"
          >
            (e.g Power outage for over 3 weeks around Aba market)
          </small>
          <input
            className="form-element"
            id="title"
            name="title"
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-wrapper no-margin-btm"
        >
          <label
            htmlFor="comments"
          >
            Comment
          </label>
          <small
            className="text-helper"
          >
            (Write full details here)
          </small>
          <textarea
            className="form-element"
            id="comments"
            name="comment"
            required={true}
          />
        </div>
      </div>
      <div
        className="report-item card card-sm max-width-800 create-report-details padding-zero-btm"
      >
        <h2
          className="report-box-title"
        >
          Report Type
        </h2>
        <p>
          Is this incident linked to corruption or just an issue that needs government intervention ?
        </p>
        <div
          className="row"
          data-pg-name="Row"
        >
          <div
            className="column column-md-6"
            data-pg-name="Column"
          >
            <div
              className="report-item max-width-800 report-box card no-margin-btm"
              title="Red Flag Report"
            >
              <h4>
                Corruption case
                 
                <i
                  className="fa fa-flag flag"
                  title="Red Flag Report"
                />
              </h4>
              <p>
                e.g. bribery, extortion, money laundering, e.t.c
              </p>
              <input
                className="radio-fine"
                data-pg-hidden={true}
                defaultChecked={true}
                defaultValue="red-flag"
                name="type"
                type="radio"
              />
              <div
                className="pg-empty-placeholder radio-mark"
              />
            </div>
          </div>
          <div
            className="column column-md-6"
            data-pg-name="Column"
          >
            <div
              className="report-item max-width-800 report-box card"
              title="Intervention Report"
            >
              <h4>
                Intervention Case
                 
                <i
                  className="fa flag fa-bullhorn flag-intervention"
                  title="Intervention Report"
                />
              </h4>
              <p>
                e.g. bad road, collapsed bridge, flooding, e.t.c
              </p>
              <input
                className="radio-fine"
                data-pg-hidden={true}
                defaultChecked={true}
                defaultValue="intervention"
                name="type"
                type="radio"
              />
              <div
                className="pg-empty-placeholder radio-mark"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="report-item card card-sm max-width-800 create-report-details padding-zero-btm"
      >
        <header
          className="header-create"
        >
          <h2
            className="report-box-title"
          >
            Location
          </h2>
          <p>
            Where did this happen? Enter the latitude/longitude. You can use
             
            <Link
              replace={false}
              target="_blank"
              to="https://maps.google.com"
            >
              <strong>
                <ins>
                  Google Map
                </ins>
              </strong>
            </Link>
             
            for help.
          </p>
          <p
            className="hidden"
          >
            Where did this happen? Enter the latitude/longitude or simply use the marker to choose a location from the map
          </p>
        </header>
        <div
          className="row"
          data-pg-name="Row"
        >
          <div
            className="column-md-6 column column-xs-6"
            data-pg-name="column-md-6"
          >
            <div
              className="form-wrapper"
            >
              <label
                htmlFor="latitude"
              >
                Latitude
              </label>
              <input
                className="form-element"
                id="latitude"
                name="latitude"
                required={true}
                type="tel"
              />
            </div>
          </div>
          <div
            className="column-md-6 column column-xs-6"
            data-pg-name="column-md-6"
          >
            <div
              className="form-wrapper"
            >
              <label
                htmlFor="longitude"
              >
                Longitude
              </label>
              <input
                className="form-element"
                id="longitude"
                name="longitude"
                required={true}
                type="tel"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="btn-wrapper max-width-800"
      >
        <button
          className="btn-brand btn-250 btn-submit"
          disabled={false}
          type="submit"
        >
          <span>
            Submit Report
          </span>
        </button>
      </div>
    </form>
  </div>
</section>
`;
